### PR TITLE
Move from Travis CI to Github Actions

### PR DESCRIPTION
**Pull request by akosveres**
## Description

The pull request has the following changes:
* Remove .travis.yaml
* Add build and publish github actions workflow
* Update README.md with new badge and documentation

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Fixes issue https://github.com/openfaas/faas/issues/1585

## How Has This Been Tested?

I've done tests on my fork:
Successful build workflow https://github.com/akosveres/nats-queue-worker/runs/1365348972?check_suite_focus=true
Succesful publish workflow https://github.com/akosveres/nats-queue-worker/runs/1365361432?check_suite_focus=true

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: build
+
+on:
+  push:
+    branches: [ master ]
```
Comment (alexellis) [inline .github/workflows/build.yaml:5]: Why is this limited to master?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.11.x]
```
Comment (alexellis) [inline .github/workflows/build.yaml:13]: Should be 1.13 now, thanks

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  publish:
+    strategy:
+      matrix:
+        go-version: [1.11.x]
```
Comment (CodeCutterUK) [inline .github/workflows/publish.yaml:12]: This should probably match the other workflows with 1.13

Comment (alexellis) [reply .github/workflows/publish.yaml:12]: Yes please

**Comment by alexellis** [comment]
Have you tested a release build on your own account? Most that I merged yesterday failed leaving broken tags

File: .github/workflows/publish.yaml (hunk 3 above)
Comment (akosveres) [reply .github/workflows/publish.yaml]: Updated

**Comment by akosveres** [comment]
> Have you tested a release build on your own account? Most that I merged yesterday failed leaving broken tags

I have as much as I could, in this repository we only take the git tag, not the sha and that looks correct.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: build
+
+on:
+  push:
+    branches: "*"
+  pull_request:
+    branches: [ master ]
```
Comment (CodeCutterUK) [inline .github/workflows/build.yaml:7]: This should be "*" based on the discussions Slack

**Comment by alexellis** [comment]
@akosveres do you have a moment to revisit the comments today?

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.13.x]
```
Comment (alexellis) [inline .github/workflows/build.yaml:13]: Do we need Go on the host?

**Comment by alexellis** [comment]
This PR is required to complete our migration, so I'm closing due to inactivity.

@utsavanand2 please can you take over in a new PR and make this work like the faas-netes or faas build please?

Pay attention to how we set the version and commit information in the image and the GitHub Action.

**Pull request closed without merging**